stablehlo/dialect/ChloOps.cpp

```diff
@@ -513,6 +514,29 @@ LogicalResult ConstantOp::inferReturnTypes(
 namespace mlir {
 namespace chlo {
 
+namespace {
+struct ChloDialectInlinerInterface : public DialectInlinerInterface {
+  using DialectInlinerInterface::DialectInlinerInterface;
+
+  // Allow all call operations to be inlined.
+  bool isLegalToInline(Operation* call, Operation* callable,
+                       bool wouldBeCloned) const final {
+    return true;
+  }
+  // We don't have any special restrictions on what can be inlined into
+  // destination regions (e.g. while/conditional bodies). Always allow it.
+  bool isLegalToInline(Region* dest, Region* src, bool wouldBeCloned,
+                       BlockAndValueMapping& valueMapping) const final {
+    return true;
+  }
+  // Operations in CHLO dialect are always legal to inline since they are pure.
+  bool isLegalToInline(Operation*, Region*, bool,
```

Comment:
Not sure if clang-tidy complains about unused variables in inlined class method definitions?

Reply:
Hmm this is what we're doing in StableHLO right now (I copypasted the implementation from there). If this becomes a problem during integrate, I'll get back and fix it here.